external/types/pdf.d.ts

```diff
@@ -78,6 +78,8 @@ declare module 'pdfjs-dist' {
 
   //
   // web/pdf_viewer.component.js
+  // Warning: These functions are only accessible with web/pdf_viewer.js.
+  //          It is NOT automatically imported with 'pdfjs-dist'.
```

Comment:
Unfortunately this is simply *not* correct, since the variables added below are not exposed *anywhere*! And certainly not on the, [soon to be removed](https://github.com/mozilla/pdf.js/projects/5), global `PDFJS` object.

The variables added here correspond to https://github.com/mozilla/pdf.js/blob/master/web/default_preferences.json, which are *only* ever used in the default viewer (i.e. the one found at https://mozilla.github.io/pdf.js/web/viewer.html) and nowhere else. These are *not* used/exposed either through the PDF.js API, nor through https://github.com/mozilla/pdf.js/blob/master/web/pdf_viewer.component.js. Hence the default viewer, i.e. file `web/app.js`, shouldn't be part of any "public" API.

Reply:
I am using some of these variables from the PDFJS object, so they have to be exposed.
Some are for example used in your examples:
https://github.com/mozilla/pdf.js/tree/master/examples/mobile-viewer

Reply:
> Some are for example used in your examples:
> https://github.com/mozilla/pdf.js/tree/master/examples/mobile-viewer

That example is *not* in any way utilizing https://github.com/mozilla/pdf.js/blob/master/web/default_preferences.json, either directly or indirectly.
I'm not sure exactly which ones that you claim is used, but if you mean https://github.com/mozilla/pdf.js/blob/ea3d8450d218dd53619c72c934547fe57f22530f/examples/mobile-viewer/viewer.js#L24-L25 then those are already covered by
https://github.com/mozilla/pdf.js/blob/ea3d8450d218dd53619c72c934547fe57f22530f/web/ui_utils.js#L76-L81
and
https://github.com/mozilla/pdf.js/blob/ea3d8450d218dd53619c72c934547fe57f22530f/web/ui_utils.js#L98-L103

Reply:
You said

> the variables added below (in this commit) are not exposed anywhere!

`useOnlyCssZoom` and `disableTextLayer` are part of those variables.
I don't get how `ui_utils.js` covers these. It only makes sure that the variables are not undefined.

Reply:
Ok, I might just have understood something. Can it be that some of those variables should not be accessible but somehow still are? The TextLayer, for example, should be activated through creating a new TextLayerBuilder and adding it to the PDFViewer?

If that is so, I just had the biggest aha moment.
And this would mean good Definition Files would really help to document all that stuff.